src/main/java/org.wso2.apim.monetization/impl/StripeMonetizationImpl.java

```diff
@@ -606,150 +631,217 @@ public boolean publishMonetizationUsageRecords(MonetizationUsagePublishInfo last
-                                    //throw MonetizationException as it will be logged and handled by the caller
-                                    throw new MonetizationException(errorMessage);
-                                }
-                            } catch (APIManagementException e) {
-                                String errorMessage = "Failed to get the Stripe key of the connected account from "
-                                        + "the : " + apiName;
+        JSONObject successAPIUsageByAppFilter = new JSONObject();
+        successAPIUsageByAppFilter.put(StripeMonetizationConstants.API_ID_COL, monetizedAPIIds);
+        successAPIUsageByAppFilter.put(StripeMonetizationConstants.TENANT_DOMAIN_COL, tenantDomains);
+        JSONObject variables = new JSONObject();
+        variables.put(StripeMonetizationConstants.TIME_FILTER, timeFilter);
+        variables.put(StripeMonetizationConstants.API_USAGE_BY_APP_FILTER, successAPIUsageByAppFilter);
+        GraphQLClient graphQLCliet =
+                Feign.builder().client(new OkHttpClient()).encoder(new GsonEncoder()).decoder(new GsonDecoder())
+                        .logger(new Slf4jLogger()).requestInterceptor(new QueyAPIAccessTokenInterceptor(accessToken))
+                        .target(GraphQLClient.class, queryApiEndpoint);
+        GraphqlQueryModel queryModel = new GraphqlQueryModel();
+        queryModel.setQuery(graphQLquery);
+        queryModel.setVariables(variables);
+        graphQLResponseClient usageResponse = graphQLCliet.getSuccessAPIsUsageByApplications(queryModel);
+        LinkedTreeMap<String, ArrayList<LinkedTreeMap<String, String>>> data = usageResponse.getData();
+        for (Map.Entry<String, ArrayList<LinkedTreeMap<String, String>>> entry : data.entrySet()) {
+            String key = entry.getKey();
+            ArrayList<LinkedTreeMap<String, String>> apiUsageDataCollection = entry.getValue();
+            for (LinkedTreeMap<String, String> apiUsageData : apiUsageDataCollection) {
+                apiName = apiUsageData.get(StripeMonetizationConstants.API_NAME);
+                apiVersion = apiUsageData.get(StripeMonetizationConstants.API_VERSION);
+                tenantDomain = apiUsageData.get(StripeMonetizationConstants.TENANT_DOMAIN);
+                applicationName = apiUsageData.get(StripeMonetizationConstants.APPLICATION_NAME);
+                applicationOwner = apiUsageData.get(StripeMonetizationConstants.APPLICATION_OWNER);
+                try {
+                    applicationId = apiMgtDAO.getApplicationId(applicationName, applicationOwner);
+                } catch (APIManagementException e) {
+                    throw new MonetizationException("Error while retrieving Application Id for " +
+                            "Applictaion " + applicationName, e);
+                }
+                requestCount = Long.parseLong(apiUsageData.get(StripeMonetizationConstants.COUNT));
```

Comment:
The entire publishMonetizationUsageRecords is lengthy. And there are several things we do inside this. We need break this upto smaller methods.

Reply:
Since this is a structural change , this and few other known improvements will be addressed in a separate PR once this is merged for the release.